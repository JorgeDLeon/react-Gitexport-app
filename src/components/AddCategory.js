import React, { useState } from 'react';
import PropTypes from 'prop-types';


function AddCategory({setCategorias}) {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {setInputValue(e.target.value)    };


    const handleSubmit = (e) =>{
        e.preventDefault();
        console.log('Handle sumit', inputValue);

        if(inputValue.trim().length > 2){
            setCategorias(cats => [inputValue, ...cats]);
            setInputValue('');
        }
    }

  return (
        <form onSubmit={ handleSubmit } >
            <p>{inputValue}</p>
            <input
                type="text" 
                value = {inputValue}
                onChange={ handleInputChange }
            />
        </form>
        
    )
}

AddCategory.protoType = {
    setCategorias: PropTypes.func.isRequired
}

export default AddCategory;


