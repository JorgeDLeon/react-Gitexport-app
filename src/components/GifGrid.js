import React, { useEffect, useState } from 'react';
import { getGifs } from '../helpers/getGifs';
import {GifGridItem} from './GifGridItem';

const GifGrid = ({categoria}) => {

    const [imagen, setImagen] = useState([]);    

    useEffect(()=>{
        getGifs(categoria)
            .then(imgs =>setImagen(imgs))
    },[categoria]) //aL Usar useEffect solo se ejecutara una sola vez el codigo que este dentro de el
    
    

  return (
    <>
        <h3 className='animate__animated animate__slideInDown'>{categoria}</h3>
        <div className='card-grid'>

              {
                imagen.map( img => (
                    <GifGridItem 
                    key={ img.id }
                    {...img} //Si se manda los parametos de esta manera se puden desestructurar en el siguiente componente
                    />
                ))
              }
          
        </div>
    </>
  )
};

export default GifGrid;

