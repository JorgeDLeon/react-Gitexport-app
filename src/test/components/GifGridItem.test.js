import React from 'react';
import { shallow } from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem';

describe('Pruebas con <GifExpertApp />', ()=>{

    const title = 'Esto es el titulo';
    const url = 'https://localhost/imagen.png';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    
    test('Debe de mostrar los componentes correctamente', ()=>{
        
        expect(wrapper).toMatchSnapshot(); //Esto tomara una fotografia del componente
        
    });

    test('Debe de tener un parrafo con el title', ()=>{

        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);

    });


    test('Debe de contener un url y title', ()=>{
        const img = wrapper.find('img');
        // console.log(img.props().src); //Esta es la manera de acceder a las propiedades de la etiqueta img
        //Tambien se pude hacer de esta manera si es solo una propiedad img.prop('src');
         // img.html() De esta manera puedo ver toda la etiqyeta img

         expect( img.prop('src') ).toBe(url);
         expect( img.prop('alt') ).toBe(title);

    });

    test('Debe de tener animate_fadeIn',()=>{

        const div = wrapper.find('div');
        const clase = div.prop('className'); //De esta manera acedemos a la clase del div
        

        expect( clase.includes('animate__fadeIn')).toBe(true); //Podemos hacer una negacion si agregamos not antes del .toBe
    
    });

})