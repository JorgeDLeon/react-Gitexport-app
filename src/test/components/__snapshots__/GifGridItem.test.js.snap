// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <GifExpertApp /> Debe de mostrar los componentes correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Esto es el titulo"
    src="https://localhost/imagen.png"
  />
  <p>
    Esto es el titulo
  </p>
</div>
`;
